public/app/features/plugins/sandbox/distortion_map.ts

```diff
@@ -520,13 +521,30 @@ async function distortPostMessage(distortions: DistortionMap) {
 }
 
 /**
- * We define "live" APIs as APIs that can only be distorted in runtime on-the-fly and not at initialization
- * time like other distortions do.
- *
- * This could be because the objects we want to patch only become available after specific states are reached
- * or because the libraries we want to patch are lazy-loaded and we don't have access to their definitions
- *
+ * "Live" APIs are APIs that can only be distorted at runtime.
+ * This could be because the objects we want to patch only become available after specific states are reached,
+ * or because the libraries we want to patch are lazy-loaded and we don't have access to their definitions.
+ * We put here only distortions that can't be static because they are dynamicly loaded
  */
-export async function distortLiveApis() {
+export function distortLiveApis(originalValue: ProxyTarget): ProxyTarget | undefined {
   distortMonacoEditor(generalDistortionMap);
+
+  // This distorts the `history.replace` function in react-router-dom.
+  // constructed for each browser history and is only accessible within the react context.
+  // Note that this distortion does not affect `String.prototype.replace` calls.
```

Comment:
What is preventing the distortion below to jump on the `String.prototype.replace` calls? Is it the `originalValue.prototype.constructor.length === 2` part?

Reply:
Calls to Js intrinsic apis such as string don't go through distortions. Let me add that in the comment to make it clearer